File: app/src/main/java/net/ducksmanager/persistence/dao/CoverSearchIssueDao.java
package net.ducksmanager.persistence.dao;

import net.ducksmanager.persistence.models.composite.CoverSearchIssue;
import net.ducksmanager.persistence.models.composite.CoverSearchIssueWithUserIssueDetails;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface CoverSearchIssueDao {
    @Query("SELECT coversearch_issue.*, issues.* FROM coversearch_issue LEFT JOIN issues ON coversearch_issue.publicationCode = issues.country || '/' || issues.magazine AND coversearch_issue.issueNumber = issues.number")
    LiveData<List<CoverSearchIssueWithUserIssueDetails>> findAll();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertList(List<CoverSearchIssue> coverSearchIssues);

    @Query("DELETE FROM coversearch_issue")
    void deleteAll();
}

File: app/src/main/java/net/ducksmanager/persistence/dao/CoverSearchIssueDao.kt
package net.ducksmanager.persistence.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import net.ducksmanager.persistence.models.composite.CoverSearchIssue
import net.ducksmanager.persistence.models.composite.CoverSearchIssueWithUserIssueAndScore

@Dao
interface CoverSearchIssueDao {
    @Query(
        " SELECT coversearch_issue.*, issues.*, suggested_issues.suggestionScore" +
            " FROM coversearch_issue" +
            " LEFT JOIN issues ON coversearch_issue.coverPublicationCode = issues.country || '/' || issues.magazine AND coversearch_issue.coverIssueNumber = issues.issueNumber" +
            " LEFT JOIN suggested_issues ON coversearch_issue.coverPublicationCode = suggested_issues.publicationCode AND coversearch_issue.coverIssueNumber = suggested_issues.issueNumber")
    fun findAll(): LiveData<List<CoverSearchIssueWithUserIssueAndScore>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertList(coverSearchIssues: List<CoverSearchIssue>)

    @Query("DELETE FROM coversearch_issue")
    fun deleteAll()
}